src/utils/generator.js

```diff
@@ -698,10 +701,29 @@ export class Generator {
     }
 
     for (let modifiedBinding of modifiedBindings.keys()) {
-      // TODO #2430: Instead of looking at the environment ids, keep instead track of a createdEnvironmentRecords set,
-      // and only consider bindings here from environment records that already existed, or even better,
-      // ensure upstream that only such bindings are ever added to the modified-bindings set.
-      if (modifiedBinding.environment.id >= environmentRecordIdAfterGlobalCode) continue;
+      // TODO: Fix this logic for React Components.
```

Comment:
The whole TODO comment around environment ids is still applicable (`environmentRecordIdAfterGlobalCode` was only ever meant as a hack; do you think it's the right thing to do now?).